File: distros/J/JavonetPerlSdk/lib/Javonet/Binaries/Python/javonet/core/generator/handler/shared/SharedClassInstanceHandler.py
from javonet.core.generator.handler.AbstractGeneretatorHandler import AbstractGeneratorHandler


class SharedClassInstanceHandler(AbstractGeneratorHandler):

    def generate_command(self, analyzed_object, parent_command, handlers):
        return analyzed_object

    def generate_code(self, existing_string_builder, common_command, used_object, handlers):
        existing_string_builder.append("\n")
        existing_string_builder.append("    instance = None")
        existing_string_builder.append("\n")

File: distros/J/JavonetPerlSdk/lib/Javonet/Binaries/Nodejs/core/generator/handler/shared/SharedClassInstanceHandler.js
const AbstractGeneratorHandler = require("../AbstractGeneratorHandler");
const OS = require("os");

class SharedClassInstanceHandler extends AbstractGeneratorHandler {

    generate_code(existing_string_builder, common_command, used_object, handlers) {
        existing_string_builder.append(OS.EOL)
        existing_string_builder.append("instance = null")
        existing_string_builder.append(OS.EOL)
    }
}

module.exports = SharedClassInstanceHandler